lib/forspell/loaders/ruby.rb

```diff
@@ -0,0 +1,62 @@
+# frozen_string_literal: true
+
+require 'yard'
+
+module Forspell::Loaders
+  class Ruby < Base
+    def initialize(file: nil)
+      super
+      @parser_class = YARD::Parser::Ruby::RubyParser
+    end
+
+    def load_comments
+      @parser_class.new(@input, @file).parse
+                   .tokens.select { |token| token.first == :comment }
+    rescue YARD::Parser::ParserSyntaxError, RuntimeError => e
+      raise Forspell::Loader::ParsingError, e.message
+      # example: [:comment, "# def loader_class path\n", [85, 2356]]
+    end
+
+    def load_words
+      inputs = []
+
+      load_comments.each do |comment|
```

Comment:
BTW, if the structure is `[:comment, "# def loader_class path\n", [85, 2356]]`, you can here
```ruby
each do |_, text, (start, fin)| # drop :comment, store text, and unpack [85, 2356] into to variables
```